### DISCUSSION TITLE
Custom Input Generator

### DISCUSSION THREAD

**Discussion by oljimenez** [chrishoermann/zod-prisma-types#92] [Ideas]
Hi, can there be a way to generate less input/output code? In my project, I only use findMany/findUnique, but I still generate a lot of unused code. I think you could add some options to the generator to select which input/output functions to generate and you could also add a field validator to omit an entire model, something like:
 
 ```prisma 

generator zod {
  provider                  = "zod-prisma-types"
  output                    = "./generated/zod"
  useMultipleFiles          = true
  createInputTypes          = true
  // this is an idea of options
  generateUpdate            = false 
}

 /// @zod.custom.omit(["model"])
model Example {
  ...
}

 ```

**Comment by chrishoermann** [comment]
@oljimenez in issue #89 we talked a bit about this, but currently I'm not planning on implementing this feature. In my experience it would only be possible for `modeltypes` and `argTypes` because the `inputTypes` heavily rely on each other and are embedded in the `prisma.dmmf` in a way that makes it hard to omit a single model.

I' will look into this when I have some more free time and see what'S possible
